src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs

```diff
@@ -32,8 +32,9 @@ public static class DotNetDispatcher
         /// <param name="jsRuntime">The <see cref="JSRuntime"/>.</param>
         /// <param name="invocationInfo">The <see cref="DotNetInvocationInfo"/>.</param>
         /// <param name="argsJson">A JSON representation of the parameters.</param>
-        /// <returns>A JSON representation of the return value, or null.</returns>
-        public static string? Invoke(JSRuntime jsRuntime, in DotNetInvocationInfo invocationInfo, string argsJson)
+        /// <param name="byteArrays">Byte array data extracted from the arguments for direct transfer.</param>
+        /// <returns>A tuple containing the JSON representation of the return value, or null, along with the extracted byte arrays, or null.</returns>
+        public static (string?, byte[][]?) Invoke(JSRuntime jsRuntime, in DotNetInvocationInfo invocationInfo, string argsJson, byte[][]? byteArrays)
```

Comment:
Might it be preferable to define a `readonly struct` for this return type rather than the tuple? Just in case, in the future, we want to add even more things.

Reply:
Agree, we don't use tuples in public APIs.

@SteveSandersonMS what was the reason for this to be public?

Reply:
> what was the reason for this to be public?

Layering reasons. JS interop is an independent layer from any of the hosting platforms, and any hosting platform that needs to deliver an inbound synchronous JS interop call would do so by calling this. Currently that's only WebAssembly.

Reply:
Thanks, updated. 
https://github.com/dotnet/aspnetcore/pull/32259/commits/3056f468a08de25362eaea98c37954c177725835#diff-d507946c1cb83ccdd4ef4e3dc431556cf387d61cb9c6e9fc6cc7c744f3868c10R10